source/reference/minio-server/settings/core.rst

```diff
@@ -371,12 +371,7 @@ Maximum Object Versions
 
 *Optional*
 
-Overrides the default maximum version per object limit of ``10000`` with the user specified value.
-
-.. important::
-
-   The default limit of 10,000 provides a safety valve against incorrect or inefficient application behavior in versioned buckets.
-   Lifting this limit without first ensuring your applications are designed for versioned operations may result in a negative performance impact over time.
-
-
-   
+Defines the default maximum versions to allow per object.
+If not defined, MinIO allows a practically unlimited number of object versions.
+ 
+The real maximum is up to highest value of ``MaxInt64``, which is over 9.2 quintillion versions per object.
```

Comment:
So there aren't performance considerations besides disk usage? We do say elsewhere that lots of versions take up lots of space, so this should not be surprising.

Reply:
we should advice that this is to avoid any breaking change upon upgrade, but you should consider setting this value to 100 or something as there is no practical reason to go beyond 100 versions per object.

Reply:
Added a line about considering 100 versions as a limit.